src/main.ts

```diff
@@ -21,32 +21,38 @@ export async function run(): Promise<void> {
       run_id: github.context.runId,
       attempt_number
     })
-
     const started_at = response.data.run_started_at
     if (!started_at) {
       throw new Error('Could not get the start time of the workflow')
     }
     const now = new Date().toISOString()
 
-    const jobs = await octokit.rest.actions.listJobsForWorkflowRunAttempt({
-      owner: github.context.repo.owner,
-      repo: github.context.repo.repo,
-      run_id: github.context.runId,
-      attempt_number
-    })
+    // check if "outcome" is set as an input, effectively overwriting the detection
+    let outcome = core.getInput('outcome')
+    if (!outcome) {
+      // it "outcome" is not set, check the status of every job in the workflow
+      const jobs = await octokit.rest.actions.listJobsForWorkflowRunAttempt({
+        owner: github.context.repo.owner,
+        repo: github.context.repo.repo,
+        run_id: github.context.runId,
+        attempt_number
+      })
 
-    let failed = false
-    for (const job of jobs.data.jobs) {
-      core.info(`Job ${job.name} has conclusion: ${job.conclusion}`)
-      failed = failed || job.conclusion === 'failure'
+      let failed = false
+      for (const job of jobs.data.jobs) {
+        core.info(`Job ${job.name} has conclusion: ${job.conclusion}`)
+        failed = failed || job.conclusion === 'failure'
+      }
+      outcome = failed ? 'failure' : 'success'
     }
```

Comment:
Another option here might have been to just pass the value of the `needs` context as an input `relevant_jobs` where we only restrict the jobs that get checked instead of setting the whole outcome manually. Like this, approximately:
```ts
      const jobs = await octokit.rest.actions.listJobsForWorkflowRunAttempt({
        owner: github.context.repo.owner,
        repo: github.context.repo.repo,
        run_id: github.context.runId,
        attempt_number
      })
      let relevant_jobs = core.getInput('relevant_jobs')

      let failed = false
      for (const job of jobs.data.jobs) {
        core.info(`Job ${job.name} has conclusion: ${job.conclusion}`)
        failed = failed || (job.conclusion === 'failure' && relevant_jobs.contains(job.name))
      }
      outcome = failed ? 'failure' : 'success'
```

But this option is fine as well.

Reply:
Yeah, I thought about that, but then I would have to build an expression which extracts the names of the jobs from the `needs` context, and then I thought this might enable more use cases in the future...